bioio/bio_image.py

```diff
@@ -164,9 +164,6 @@ def determine_plugin(
 
             # Check for extension in plugins_by_ext
             for format_ext, plugins in plugins_by_ext.items():
-                # Remove "." prefix if already there
-                if format_ext.startswith("."):
-                    format_ext = format_ext[1:]
```

Comment:
Moved to plugins.py as without it, duplicate suffixes are stored in the plugin cache

Reply:
this one-by-one linear search kinda sucks but I don't have any better ideas for how to do it (the following line after this change).  I guess generally it's better to just specify the Reader up front :)

Reply:
As a nitpick I would also like to push for having it always _add_ the `.` at the front rather than remove it. As we are already doing that a line below with:

`if path.lower().endswith(f".{format_ext}"):`

This line is also making me realize that we need to lowercase all of the ext in the plugin cache as well (there are already plugins (bioio-imageio) which are reporting capitalized suffixes).

I can do those two things in another PR. Just would love your quick thoughts. It just seems like we should be doing as much of the extension prep and manipulation in the plugin cache creation rather than at `determine_plugin` time.